tools/packaging/packaging.psm1

```diff
@@ -246,255 +260,305 @@ function New-UnixPackage {
-        $symlink_dest = "$Destination/$Name"
-        $hack_dest = "./_fpm_symlink_hack_powershell"
-        if ($Environment.IsMacOS) {
-            if (Test-Path $symlink_dest) {
-                Write-Warning "Move $symlink_dest to $hack_dest (fpm utime bug)"
-                Move-Item $symlink_dest $hack_dest
+            # run ronn to convert man page to roff
+            $RonnFile = Join-Path $PSScriptRoot "/../../assets/powershell.1.ronn"
+            $RoffFile = $RonnFile -replace "\.ronn$"
+
+            # Run ronn on assets file
+            # Run does not play well with files named powershell6.0.1, so we generate and then rename
+            Start-NativeExecution { ronn --roff $RonnFile }
+
+            # Setup for side-by-side man pages (noop if primary package)
+            $FixedRoffFile = $RoffFile -replace "powershell.1$", "$Name.1"
+            if ($Name -ne "powershell") {
+                Move-Item $RoffFile $FixedRoffFile
             }
-        }
 
-        # run ronn to convert man page to roff
-        $RonnFile = Join-Path $PSScriptRoot "/../../assets/powershell.1.ronn"
-        $RoffFile = $RonnFile -replace "\.ronn$"
+            # gzip in assets directory
+            $GzipFile = "$FixedRoffFile.gz"
+            Start-NativeExecution { gzip -f $FixedRoffFile }
 
-        # Run ronn on assets file
-        # Run does not play well with files named powershell6.0.1, so we generate and then rename
-        Start-NativeExecution { ronn --roff $RonnFile }
+            $ManFile = Join-Path "/usr/local/share/man/man1" (Split-Path -Leaf $GzipFile)
 
-        # Setup for side-by-side man pages (noop if primary package)
-        $FixedRoffFile = $RoffFile -replace "powershell.1$", "$Name.1"
-        if ($Name -ne "powershell") {
-            Move-Item $RoffFile $FixedRoffFile
+            # Change permissions for packaging
+            Start-NativeExecution {
+                find $Staging -type d | xargs chmod 755
+                find $Staging -type f | xargs chmod 644
+                chmod 644 $GzipFile
+                chmod 755 "$Staging/$Name" # only the executable should be executable
+            }
         }
 
-        # gzip in assets directory
-        $GzipFile = "$FixedRoffFile.gz"
-        Start-NativeExecution { gzip -f $FixedRoffFile }
-
-        $ManFile = Join-Path "/usr/local/share/man/man1" (Split-Path -Leaf $GzipFile)
+        # Setup package dependencies
+        # These should match those in the Dockerfiles, but exclude tools like Git, which, and curl
+        $Dependencies = @()
+        if ($Environment.IsUbuntu -or $Environment.IsDebian) {
+            $Dependencies = @(
+                "libc6",
+                "libcurl3",
+                "libgcc1",
+                "libgssapi-krb5-2",
+                "liblttng-ust0",
+                "libstdc++6",
+                "libunwind8",
+                "libuuid1",
+                "zlib1g"
+            )
+
+            if ($DebDistro -eq "ubuntu.14.04") {
```

Comment:
would this be more readable as a switch statement?

Reply:
possible. I will use `switch` instead.